### DISCUSSION TITLE
Provide collapse groups for composite steps

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#15002] [Actions]
I'm transitioning from a multistep workflow job with a single-step composite action to a single-step workflow job with a multi-step composite action.

When one uses multiple steps in a workflow, each step gets its own expando widget, and generally only the interesting step is expanded.

When I switched, while the GHA implementation details for retrieving the actions used in the composite steps are grouped (and collapsed), the general output from the steps are inlined w/ the rest of my action.

https://github.com/check-spelling/openrc/runs/6039482591?check_suite_focus=true#step:2:550

<img width="927" alt="image" src="https://user-images.githubusercontent.com/2119212/163586846-0ccdb9b3-b3fe-4997-83bc-c7510a71f8eb.png">

Here, `▶Run actions/upload-artifact@v3` is the beginning of a new [step in the composite action](https://github.com/check-spelling/check-spelling/blob/b07033951b80d941bb0d420183473509e10fd67c/action.yml#L261)

I'd love for the action phases to be treated like this instead:

<details><summary>actions/upload-artifact@v3 </summary>

<details><summary>Run actions/upload-artifact@v3</summary>

```yaml
  with:
    retention-days: 1
    name: check-spelling-comment-2173068206
    path: /tmp/tmp.nG6UAYmSOj
  
    if-no-files-found: warn
  env:
    result_code: 1
```
</details>

```
With the provided path, there will be 5 files uploaded
Starting artifact upload
For more detailed logs during the artifact upload process, enable step-debugging: https://docs.github.com/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging#enabling-step-debug-logging
Artifact name is valid!
Container for artifact "check-spelling-comment-2173068206" successfully created. Starting upload of file(s)
Total size of all the files uploaded is 132 bytes
File upload process has finished. Finalizing the artifact upload
Artifact has been finalized. All files have been successfully uploaded!

The raw size of all the files that were specified for upload is 132 bytes
The size of all the files that were uploaded is 132 bytes. This takes into account any gzip compression used to reduce the upload size, time and storage

Note: The size of downloaded zips can differ significantly from the reported size. For more information see: https://github.com/actions/upload-artifact#zipped-artifact-downloads 

Artifact check-spelling-comment-2173068206 has been successfully uploaded!
```
</details>

**Comment by Ugzuzg** [comment]
There was also a great suggestion in the issues, but was closed with a request to move it to the discussions: https://github.com/actions/runner/issues/1311
![Collapsable composite action steps inside top-level action](https://user-images.githubusercontent.com/2366731/132258346-deaba1f2-2b8e-49ef-83c3-b98c9bf89533.png)

**Reply by osievert-gpfw** [reply]
Couple years later now but the opportunity for improvement still exists here. By visually obfuscating individual steps in a composite action, it really ties our hands in how we construct workflows and actions. We can't collect related steps into a composite action which simplifies the top-level workflow without paying a significant readability cost when we do it. It is A LOT easier to triage and debug steps in a workflow than steps in a composite action. This leads us all to favor workflows with a lot of small steps, and to disfavor composite actions, both of which hurt our ability to create simple workflows with modular and reusable pieces.

To improve this, nothing needs to change about the data model or behavior of github actions. Just a bit of visual design effort to display composite steps more distinctly in the job output. Easy to do, big win for the usability of the platform.
